collectors/sentinelone/test/sentinelone_test.js

```diff
@@ -0,0 +1,256 @@
+                poll_interval_sec: 1
+            };
+            const nextPage = "cursor";
+            SentineloneCollector.load().then(function (creds) {
+                var collector = new SentineloneCollector(ctx, creds, 'sentinelone');
+                let nextState = collector._getNextCollectionStateWithNextPage(curState, nextPage);
+                assert.ok(nextState.nextPage);
+                assert.equal(nextState.nextPage, nextPage);
+                done();
+            });
+        });
+    });
+
+    describe('Health Check Tests', function () {
+        it('token validation check success', function (done) {
+            const sentineloneHealth = require('../health_checks');
+            alserviceStuPost = sinon.stub(RestServiceClient.prototype, 'post').callsFake(
+                function fakeFn(path, extraOptions) {
+                    return new Promise(function (resolve, reject) {
+                        assert.notEqual(path, '/web/api/v2.0/users/generate-api-token');
+                        return resolve({ data: { expiresAt: moment().add(5, 'days').toISOString() } });
+                    });
+                });
+
+            sentineloneHealth.sentinelOneTokenHealthCheck(function (err) {
+                assert.equal(null, err);
+                alserviceStuPost.restore();
+                done();
+            });
+
+        });
+        it('token expire check success', function (done) {
+            const sentineloneHealth = require('../health_checks');
+            let responseArray = [];
+            alserviceStuPost = sinon.stub(RestServiceClient.prototype, 'post').callsFake(
+                function fakeFn(path, extraOptions) {
+                    return new Promise(function (resolve, reject) {
+                        var response = null;
+                        switch (path) {
+                            case '/web/api/v2.0/users/api-token-details':
+                                response = { data: { expiresAt: moment().subtract(2, 'days').toISOString() } };
+                                responseArray.push(response);
+                                break;
+                            case '/web/api/v2.0/users/generate-api-token':
+                                response = { data: { token: "token" } };
+                                responseArray.push(response);
+                                break;
+                            default:
+                                break;
+                        }
+                        return resolve(response);
+                    });
+                });
+
+            let targetObject = {
+                setPawsSecret(token) {
+                    return Promise.resolve('success');
+                }
+            };
+            sentineloneHealth.sentinelOneTokenHealthCheck.bind(targetObject)(function (err) {
+                assert(responseArray.length == 2, "Response Array length is wrong");
```

Comment:
You don't need to do it this way. you can just determine how many times the stub was called with `callCount`

https://sinonjs.org/releases/latest/spies/